components/channel_header/channel_header.js

```diff
@@ -0,0 +1,577 @@
+} from 'utils/constants';
+import * as Utils from 'utils/utils';
+
+import ChannelHeaderPlug from 'plugins/channel_header_plug';
+
+import HeaderIconWrapper from './components/header_icon_wrapper';
+
+const headerMarkdownOptions = {singleline: true, mentionHighlight: false, atMentions: true};
+const popoverMarkdownOptions = {singleline: false, mentionHighlight: false, atMentions: true};
+
+const SEARCH_BAR_MINIMUM_WINDOW_SIZE = 1140;
+
+export default class ChannelHeader extends React.PureComponent {
+    static propTypes = {
+        teamId: PropTypes.string.isRequired,
+        currentUser: PropTypes.object.isRequired,
+        channel: PropTypes.object,
+        channelMember: PropTypes.object,
+        dmUser: PropTypes.object,
+        isFavorite: PropTypes.bool,
+        isReadOnly: PropTypes.bool,
+        isMuted: PropTypes.bool,
+        rhsState: PropTypes.oneOf(
+            Object.values(RHSStates),
+        ),
+        actions: PropTypes.shape({
+            favoriteChannel: PropTypes.func.isRequired,
+            unfavoriteChannel: PropTypes.func.isRequired,
+            showFlaggedPosts: PropTypes.func.isRequired,
+            showPinnedPosts: PropTypes.func.isRequired,
+            showMentions: PropTypes.func.isRequired,
+            closeRightHandSide: PropTypes.func.isRequired,
+            updateRhsState: PropTypes.func.isRequired,
+            getCustomEmojisInText: PropTypes.func.isRequired,
+            updateChannelNotifyProps: PropTypes.func.isRequired,
+            goToLastViewedChannel: PropTypes.func.isRequired,
+            openModal: PropTypes.func.isRequired,
+            closeModal: PropTypes.func.isRequired,
+        }).isRequired,
+    };
+
+    static defaultProps = {
+        dmUser: {},
+    };
+
+    constructor(props) {
+        super(props);
+
+        const showSearchBar = Utils.windowWidth() > SEARCH_BAR_MINIMUM_WINDOW_SIZE;
+        this.state = {
+            showSearchBar,
+            showQuickSwitch: false,
+        };
+
+        this.getHeaderMarkdownOptions = memoizeResult((channelNamesMap) => (
+            {...headerMarkdownOptions, channelNamesMap}
+        ));
+        this.getPopoverMarkdownOptions = memoizeResult((channelNamesMap) => (
+            {...popoverMarkdownOptions, channelNamesMap}
+        ));
+    }
+
+    componentDidMount() {
+        this.props.actions.getCustomEmojisInText(this.props.channel.header);
+        document.addEventListener('keydown', this.handleShortcut);
+        document.addEventListener('keydown', this.handleQuickSwitchKeyPress);
+        window.addEventListener('resize', this.handleResize);
+    }
+
+    componentWillUnmount() {
+        document.removeEventListener('keydown', this.handleShortcut);
+        document.removeEventListener('keydown', this.handleQuickSwitchKeyPress);
+        window.removeEventListener('resize', this.handleResize);
+    }
+
+    componentDidUpdate() {
+        this.props.actions.getCustomEmojisInText(this.props.channel.header);
```

Comment:
This should only happen when the header actually changes like
```javascript
componentDidUpdate(prevProps) {
    const header = this.props.channel ? this.props.channel.header : '';
    const prevHeader = prevProps.channel ? prevProps.channel.header : '';
    if (header !== prevHeader) {
        this.props.actions.getCustomEmojisInText(header);
    }
}
```
Otherwise, it'll try to get emojis even when there shouldn't be any ones to get